app/controllers/courses/modern_languages_controller.rb

```diff
@@ -77,6 +74,18 @@ def selected_language_subjects
       end
     end
 
+    def selected_non_language_subjects
+      ids = params.fetch(:course, {})[:subjects_ids]
+
+      if ids.present?
+        ids.map do |id|
+          Subject.new(id: id)
+        end
+      else
+        []
+      end
```

Comment:
If you implement saus change (https://github.com/DFE-Digital/publish-teacher-training/pull/930/files#r387766944) you can remove ths `if` and just do the map